User/UserWithPassword.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace SumoCoders\FrameworkMultiUserBundle\User;
+
+use SumoCoders\FrameworkMultiUserBundle\Security\PasswordResetToken;
+
+class UserWithPassword implements User, PasswordReset
+{
+    /**
+     * @var string
+     */
+    protected $username;
+
+    /**
+     * @var string
+     */
+    protected $password;
+
+    /**
+     * @var string
+     */
+    protected $displayName;
+
+    /**
+     * @var PasswordResetToken
+     */
+    protected $passwordResetToken;
+
+    /**
+     * @var string
+     */
+    protected $email;
+
+    /**
+     * @var int
+     */
+    protected $id;
+
+    /**
+     * @param string $username
+     * @param string $password
+     * @param string $displayName
+     * @param string $email
+     * @param int $id
+     * @param PasswordResetToken $token
+     */
+    public function __construct(
+        $username,
+        $password,
+        $displayName,
+        $email,
+        $id = null,
+        PasswordResetToken $token = null
+    ) {
+        $this->username = $username;
+        $this->password = $password;
+        $this->displayName = $displayName;
+        $this->email = $email;
+        $this->id = $id;
+
+        if ($token) {
+            $this->passwordResetToken = $token;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getRoles()
+    {
+        return [ 'ROLE_USER' ];
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getPassword()
+    {
+        return $this->password;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getSalt()
+    {
+        return;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getUsername()
+    {
+        return $this->username;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function eraseCredentials()
```

Comment:
I wrote this for the user without a password, in which case it shouldn't do a thing.

When a user has a password, this one should erase the plainPassword from the entity, only keeping the encoded version.